pkgs/by-name/zu/zulip-emoji-mapping/package.nix

```
@@ -0,0 +1,35 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3Packages,
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "zulip-emoji-mapping";
+  version = "1.0.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "GearKite";
+    repo = "zulip-emoji-mapping";
+    rev = "v${version}";
+    hash = "sha256-logm5uAnLAcFqI7mUxKEO9ZmHqRkd6CFiCW4B5tqZzg=";
+  };
+
+  build-system = with python3Packages; [
+    poetry-core
+    setuptools
+  ];
```

```suggestion
    setuptools
  ];
```
`setuptools` is needed because the upstream packages are incorrect.
You can use `setuptools` as [a fallback](https://setuptools.pypa.io/en/latest/build_meta.html#what-is-it) since `zulip-emoji-mapping` is simple enough, or you can send a patch to the upstream to add the following `build-system` to pyproject.toml and use `poetry-core`.
In any case, please leave some comments.

```
[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```
https://python-poetry.org/docs/pyproject/#poetry-and-pep-517